colcon_defaults/argument_parser/defaults.py

```diff
@@ -106,6 +123,25 @@ def collect_parsers_by_verb(root, parsers, parent_verbs=()):
 
         data = self._get_defaults_values(self._config_path)
 
+        # Additionally check for these potential config files local to the
+        # workspace. Warn and discard if multiple exist.
+        local_config_paths = [
+            Path('.colcon_defaults.yaml'),
+            Path('colcon_defaults.yaml'),
+            Path('.colcon/defaults.yaml'),
```

Comment:
At the risk of bikeshedding, I'd really like to pick just one of these. It will simplify the documentation and debugging trying to trace how a value gets set. My vote is for `colcon_defaults.yaml`, but I don't feel particularly married to it.

Reply:
If we want to allow one only then I agree the regular non-hidden file is the way to go. But is it really a notable concern to allow more than one option? I would expect some people would appreciate the hidden-file option at least.

Reply:
Just chiming in here, because I'm eager to see this PR get across the line :) 

So looking around a bit, I think colcon ignores any directory that starts with a dot when traversing the filesystem, like the [metadata](https://github.com/colcon/colcon-metadata/blob/480c4c52fb34869dab788641e7c0e33750e5b0d6/colcon_metadata/metadata/__init__.py#L42) search. That mixin by default searches for a single [./colcon_meta](https://github.com/colcon/colcon-metadata/blob/480c4c52fb34869dab788641e7c0e33750e5b0d6/colcon_metadata/package_discovery/colcon_meta.py#L42) file, but it can be modified with `--metas` and allow specification of multiple files.

Conveniently, using a defaults file with:
```
{
  "build":{
    "metas":["/tmp/foo.meta", "./colcon.meta"]
  }
}
```

I see `metas=['/tmp/foo.meta', './colcon.meta']` in the parsed commandline, and it also reads the `/tmp/foo.meta` file when I build. Given that the global defaults file from `COLCON_HOME` is loaded on line 123, could we do something similar here where the `COLCON_HOME` defaults can modify / extend the paths allowed for this file? It would be in line with the handling for the meta file, allow for a singular default path, but still allow users to hide the file if they'd wish to do so?

Reply:
@iwanders I'm making this PR as simple as possible now so we can at least get the feature merged but that sounds like an interesting idea for a followup.

Reply:
@jdlangs , yup, sounds like a good idea! Current version is functionally identical to the version we have been using ([compare](https://github.com/jdlangs/colcon-defaults/compare/feature/local-defaults...clearpathrobotics:cpr-devel-jdlangs-feature-local-defaults)), which I made from this PR around December.

Good idea to settle on a single file, it makes it much simpler. Thanks for picking this up again :+1: